chitchat/src/lib.rs

```diff
@@ -159,16 +156,17 @@ impl Chitchat {
     /// Reports heartbeats to the failure detector for nodes in the delta for which we received an
     /// update.
     fn report_heartbeats(&mut self, delta: &Delta) {
-        for (chitchat_id, node_delta) in &delta.node_deltas {
-            if let Some(node_state) = self.cluster_state.node_states.get(chitchat_id) {
-                if node_state.heartbeat() < node_delta.heartbeat
-                    || node_state.max_version() < node_delta.max_version
```

Comment:
After this patch, we only record as heartbeat an update in the actual heartbeat KV.